app-frontend/src/app/components/tools/diagramContainer/diagramContainer.controller.js

```diff
@@ -112,38 +118,55 @@ export default class DiagramContainerController {
                     on-change="onChange({override: override})"
                   ></rf-classify-node>
                   <rf-node-histogram
-                    ng-if="showHistogram && showBody"
+                    ng-if="currentView === 'HISTOGRAM'"
                     data-histogram="histogram"
                     data-breakpoints="breakpoints"
                     data-masks="masks"
                     data-options="histogramOptions"
                     on-breakpoint-change="onBreakpointChange(breakpoints, options)"
                   ></rf-node-histogram>
+                  <rf-node-statistics
+                    ng-if="currentView === 'STATISTICS'"
+                    data-model="model"
+                    data-toolrun="model.get('toolrun')"
+                    data-size="model.get('size')"
+                  ></rf-node-statistics>
                 </div>`,
             initialize: function () {
                 _.bindAll(this, 'updateBox');
                 joint.dia.ElementView.prototype.initialize.apply(this, arguments);
                 this.model.on('change', this.updateBox, this);
                 this.$box = angular.element(this.template);
                 this.scope = $scope.$new();
-                this.scope.showBody = true;
+                // Acceptable values are 'BODY', 'HISTOGRAM', and 'STATISTICS'
+                this.scope.currentView = 'BODY';
                 let dropdownHeight = 200;
                 let headerHeight = 50;
                 this.histogramHeight = dropdownHeight + headerHeight;
 
                 this.scope.toggleHistogram = () => {
-                    this.scope.showHistogram = !this.scope.showHistogram;
-                    if (this.scope.showHistogram && !this.scope.showBody) {
-                        this.scope.toggleBody();
-                    } else if (this.scope.showHistogram) {
+                    if (this.scope.currentView === 'HISTOGRAM') {
+                        this.scope.currentView = 'BODY';
+                        this.model.resize(this.expandedSize.width, this.expandedSize.height);
+                    } else {
+                        this.scope.currentView = 'HISTOGRAM';
                         this.expandedSize = this.model.getBBox();
                         this.model.resize(this.expandedSize.width, this.histogramHeight);
-                    } else if (this.scope.showBody) {
+                    }
+                };
+                this.scope.toggleStatistics = () => {
+                    if (this.scope.currentView === 'STATISTICS') {
+                        this.scope.currentView = 'BODY';
                         this.model.resize(this.expandedSize.width, this.expandedSize.height);
+                    } else {
+                        this.scope.currentView = 'STATISTICS';
+                        this.expandedSize = this.model.getBBox();
+                        this.model.resize(this.expandedSize.width, this.histogramHeight);
```

Comment:
I noticed a couple issues with the sizing functionality:
- Clicking the stats button, then the histogram, and then the histogram again will cause the node to retain the larger size from when the stats were displayed.
- Clicking the collapse arrow throws a console error the first time, and after that it sort of works but the node contents are still visible when the node is collapsed.